pkg/cmd/cli/cmd/login.go

```diff
@@ -178,19 +181,71 @@ func (o LoginOptions) Validate(args []string, serverFlag string) error {
 	return nil
 }
 
-// RunLogin contains all the necessary functionality for the OpenShift cli login command
-func RunLogin(cmd *cobra.Command, options *LoginOptions) error {
-	if err := options.GatherInfo(); err != nil {
+func (o *LoginOptions) RunLogin() error {
+	if !o.Force && o.isLikelyAdminKubeConfig() {
+		return errors.New("It looks like your config file is used for a cluster-admin (system:admin).  You probably shouldn't modify this file using `oc login`.\n" +
+			"If you really want to modify this file, use `--force`, otherwise specify a different path with --config or by setting $KUBECONFIG.")
+	}
+	if !o.Force && o.isReadOnly() {
+		return errors.New("Your config file is read only so `oc login` will probably fail.\n" +
+			"If you really want to modify this file, use `--force`, otherwise specify a different path with --config or by setting $KUBECONFIG.")
+	}
+
+	if err := o.GatherInfo(); err != nil {
 		return err
 	}
 
-	newFileCreated, err := options.SaveConfig()
+	newFileCreated, err := o.SaveConfig()
 	if err != nil {
 		return err
 	}
 
 	if newFileCreated {
-		fmt.Fprintln(options.Out, "Welcome! See 'oc help' to get started.")
+		fmt.Fprintln(o.Out, "Welcome! See 'oc help' to get started.")
 	}
 	return nil
 }
+
+func (o *LoginOptions) isLikelyAdminKubeConfig() bool {
+	if len(o.StartingKubeConfig.CurrentContext) == 0 {
+		return false
+	}
+
+	context := o.StartingKubeConfig.Contexts[o.StartingKubeConfig.CurrentContext]
+	if len(context.AuthInfo) == 0 {
+		return false
+	}
+
+	// when we generate, we always choose this name
+	if strings.HasPrefix(context.AuthInfo, "system:admin/") {
+		return true
+	}
+
+	return false
+}
+
+func (o *LoginOptions) isReadOnly() bool {
```

Comment:
Name is a little misleading, the `LoginOptions` are not read-only if this returns `true` -- `isKubeconfigReadOnly` ?